bluepyopt/ephys/create_hoc.py

```diff
@@ -33,20 +40,29 @@
     'myelinated']
 
 
-def _generate_channels_by_location(mechs, location_order):
+def _generate_channels_by_location(mechs, location_order, loc_desc):
     """Create a OrderedDictionary of all channel mechs for hoc template."""
     channels = OrderedDict((location, []) for location in location_order)
+    point_channels = OrderedDict((location, []) for location in location_order)
     for mech in mechs:
         name = mech.suffix
         for location in mech.locations:
-            # TODO this is dangerous, implicitely assumes type of location
-            channels[location.seclist_name].append(name)
-    return channels
+            if isinstance(mech, mechanisms.NrnMODPointProcessMechanism):
+                point_channels[loc_desc(location, mech)].append(mech)
+            else:
+                channels[loc_desc(location, mech)].append(name)
```

Comment:
In the else branch `mech` parameter shouldn't be passed to `loc_desc`.

Reply:
This one is necessary in both branches.